lib/bootloader_setup.pm

```diff
@@ -64,120 +78,154 @@ sub select_bootmenu_option {
         # and UEFI menues.
         send_key 'up';
     }
-    if (get_var('ZDUP') || get_var("ONLINE_MIGRATION")) {
-        send_key 'ret';    # boot from hard disk
+    if (get_var('ZDUP') || get_var('ONLINE_MIGRATION')) {
+        boot_local_disk;
         return 3;
     }
 
-    if (get_var("UPGRADE")) {
+    if (get_var('UPGRADE')) {
         # random magic numbers
-        send_key_until_needlematch('inst-onupgrade', 'down', 10, 5);
+        if (check_var('ARCH', 'ppc64le')) {
+            send_key_until_needlematch 'inst-onupgrade', 'up', 10, 5;
```

Comment:
but you are using it more than once below so please find a way to set it just once with a comment why ppc64le needs special treatment